exercises/high-scores/example.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Debug)]
+pub struct HighScores {
+    scores: Vec<u32>,
+}
+
+impl HighScores {
+    pub fn new(scores: &[u32]) -> Self {
+        Self {
+            scores: scores.to_vec(),
+        }
+    }
+
+    pub fn scores(&self) -> &[u32] {
+        &self.scores
+    }
+
+    pub fn latest(&self) -> u32 {
+        *self.scores.last().unwrap()
+    }
+
+    pub fn personal_best(&self) -> u32 {
+        *self.scores.iter().max().unwrap()
+    }
+
+    pub fn personal_top_three(&self) -> Vec<u32> {
+        let mut sorted = self.scores.clone();
+        sorted.sort();
+        sorted.iter().rev().take(3).cloned().collect()
```

Comment:
We can't avoid the first instance of cloning the list, since we otherwise have no way to sort it. Still, we can at least avoid the second clone/collect cycle.

```suggestion
        sorted.reverse();
        sorted.truncate(3);
        sorted
```

Reply:
Truncate is coll, didn't know about it. Thanks!